### PR TITLE
CLN: organize MultiIndex indexing tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/indexes/conftest.py [hunk 1]
```diff
@@ -1,8 +1,30 @@
+import numpy as np
 import pytest
 
 import pandas._testing as tm
 from pandas.core.indexes.api import Index, MultiIndex
 
+
+def _gen_mi():
+    # a MultiIndex used to test the general functionality of the
```
Comment (WillAyd) [inline pandas/tests/indexes/conftest.py]: I think "general functionality" is mistakenly repeated twice

File: pandas/tests/indexes/conftest.py [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+    # general functionality of this object
+
+    # See Also: tests.multi.conftest.idx
+    major_axis = Index(["foo", "bar", "baz", "qux"])
```
Comment (WillAyd) [inline pandas/tests/indexes/conftest.py]: Can you alternately just construct the mi from tuples? I think would be more readable that way

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor nits

File: pandas/tests/indexes/conftest.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/conftest.py]: thanks, will update

File: pandas/tests/indexes/conftest.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/conftest.py]: This is how it is constructd in the other conftest

**Review by jreback — COMMENTED** [review]
seems ok, but is there a reason to use the class based structure rather than simply splitting to modules? (and/or putting in existing modules)

e.g. TestSliceLocs we already have test_slice.py (haven't looked closely, but for example)

**Comment by jbrockmendel** [comment]
> but is there a reason to use the class based structure rather than simply splitting to modules? (and/or putting in existing modules)

habit, mostly.  for some stuff its nice to organize tests within a module

**Comment by jreback** [comment]
thanks

**Pull request merged**